File: Backend/DTOs/RegisterDto.cs
using System.ComponentModel.DataAnnotations;

namespace Backend.DTOs
{
    public class RegisterDto
    {
        [Required, StringLength(100)]
        public string Nom { get; set; } = null!;

        [Required, EmailAddress]
        public string Email { get; set; } = null!;

        [Required, MinLength(6)]
        public string Password { get; set; } = null!;

        /// <summary>
        /// Role: "Candidat" ou "Entreprise"
        /// </summary>
        [Required]
        public string Role { get; set; } = null!;

        // Champs supplémentaires pour Candidat
        public string? Prenom { get; set; }
        public DateTime? DateNaissance { get; set; }

        // Champs supplémentaires pour Entreprise
        public string? Secteur { get; set; }
    }
}

File: Backend/src/auth/dto/register.dto.ts
import {
    IsEmail,
    IsNotEmpty,
    IsIn,
    IsOptional,
    MinLength,
} from 'class-validator';

import { ApiProperty } from '@nestjs/swagger';

export class RegisterDto {
    @ApiProperty({ example: 'Doe' })
    @IsNotEmpty()
    nom: string;

    @ApiProperty({ example: 'John', required: false })
    @IsOptional()
    prenom?: string;

    @ApiProperty({ example: 'john.doe@example.com' })
    @IsEmail()
    email: string;

    @ApiProperty({ example: 'password123', minLength: 6 })
    @IsNotEmpty()
    @MinLength(6)
    password: string;

    @ApiProperty({ enum: ['Candidat', 'Entreprise'] })
    @IsIn(['Candidat', 'Entreprise'])
    role: 'Candidat' | 'Entreprise';

    @ApiProperty({ example: 'Informatique', required: false })
    @IsOptional()
    secteur?: string;

    @ApiProperty({ example: '1990-01-01', required: false })
    @IsOptional()
    dateNaissance?: string;
}
